src/utils/iceberg_type.cpp

```diff
@@ -9,12 +10,41 @@
 
 namespace duckdb {
 
+void IcebergTypeHelper::VerifyTypeValidity(CreateTableInfo &info) {
+	for (idx_t i = 0; i < info.columns.LogicalColumnCount(); i++) {
+		auto &column = info.columns.GetColumn(LogicalIndex(i));
+		switch (column.Type().id()) {
+		case LogicalTypeId::BOOLEAN:
+		case LogicalTypeId::INTEGER:
+		case LogicalTypeId::BIGINT:
+		case LogicalTypeId::FLOAT:
+		case LogicalTypeId::DOUBLE:
+		case LogicalTypeId::DECIMAL:
+		case LogicalTypeId::DATE:
+		case LogicalTypeId::TIME:
+		case LogicalTypeId::TIMESTAMP:
+		case LogicalTypeId::TIMESTAMP_TZ:
+		case LogicalTypeId::VARCHAR:
+		case LogicalTypeId::UUID:
+		case LogicalTypeId::BLOB:
+		case LogicalTypeId::ARRAY:
+		case LogicalTypeId::STRUCT:
+		case LogicalTypeId::LIST:
+		case LogicalTypeId::MAP:
+			continue;
+		default:
+			throw InvalidInputException("Column type %s is not a valid Iceberg Type.",
+			                            LogicalTypeIdToString(column.Type().id()));
+		}
+	}
+}
+
 string IcebergTypeHelper::LogicalTypeToIcebergType(const LogicalType &type) {
 	switch (type.id()) {
-	case LogicalTypeId::TINYINT:
+	case LogicalTypeId::USMALLINT:
 	case LogicalTypeId::UTINYINT:
```

Comment:
I don't understand why we are accepting some of the unsigned types?
Especially because we did remove all tests for unsigned types now